### ISSUE TITLE
Type cheking time complexity: weird behavior

### ISSUE THREAD

**Issue by nicolabotta** [idris-lang/Idris-dev#3197]
On a fresh installation of https://github.com/nicolabotta/SeqDecProbs.git, you should be able to do
```
$ cd SeqDecProbs-master/frameworks/14-/
$ time idris +RTS -K32000000 -RTS -p contrib -V --check NonNegRationalBasicTests.lidr
```
This should yield something like
```
Type checking ./Sigma.lidr
Type checking ./PairsOperations.lidr
Type checking ./NatLTEProperties.lidr
Type checking ./NatBasicProperties.lidr
Type checking ./NatLTProperties.lidr
...
Type checking ./FractionBasicProperties.lidr
...
Type checking ./NonNegRationalPredicates.lidr
Type checking ./NonNegRationalBasicProperties.lidr
Type checking ./NonNegRationalBasicTests.lidr

real	22m22.556s
user	22m15.696s
sys	0m7.596s
```
The strange thing is that the time it takes to type check individual files appears to be extremely scattered: most files will type check in a few seconds, but some files will take minutes.

For instance, it takes about 7 minutes (out of 22 minutes overall) to type check `FractionBasicProperties.lidr`. This is a bit strange but ... let it be. 

But it also takes one and an half minutes to type check `NonNegRationalPredicates.lidr` which is just a two liner:
```
> module NonNegRationalPredicates

> import NonNegRational
> import NonNegRationalBasicOperations
> import Fraction
> import FractionPredicates

> %default total
> %access public export
> %auto_implicits off


> ||| Proofs that `x` is less than or equal to `y`                                                                          
> ||| @ x the smaller number                                                                                                
> ||| @ y the larger number
> LTE : (x, y : NonNegRational) -> Type
> LTE x y = FractionPredicates.LTE (toFraction x) (toFraction y)
```
Notice also that deleting `FractionBasicProperties.ibc` after `NonNegRationalBasicTests.ibc` has been built and type checking again
```
$ rm FractionBasicProperties.ibc
$ idris +RTS -K32000000 -RTS -p contrib -V --check FractionBasicProperties.lidr
```
executes in about 90 seconds (instead of 7 minutes!). Similarly
```
$ rm NonNegRationalPredicates.ibc 
$ idris +RTS -K32000000 -RTS -p contrib -V --check NonNegRationalPredicates.lidr 
```
executes in about 10 seconds instead of one and an half minutes!

What is going on here? Am I hitting some complexity barrier of the process that keeps track of module dependencies? 

Why do certain files (even relatively long ones like, for instance, `VectProperties.lidr`) type check in a few seconds but it takes ages to type check a simple data declaration like the one in `NonNegRationalPredicates.lidr`?

Any idea how to reduce the time it takes to type check the above examples?

**Comment by ahmadsalim** [comment]
I think this would be solved by #3199 and so I hope it is OK if I close it.
If you still experience issues after #3199 is solved, we could reopen this issue.

**Comment by nicolabotta** [comment]
> I think this would be solved by #3199 and so I hope it is OK if I close it.
I think this is a very bold thinking but I very much hope you are right! Until #3199 is fixed, a simple `%unfreeze` directive should do the job, as proposed in #3200.
